inference-engine/src/legacy_api/src/net_pass.cpp

```diff
@@ -1333,10 +1333,15 @@ namespace {
 template <Precision::ePrecision PREC_FROM, Precision::ePrecision PREC_TO>
 void convertArrayPrecision(typename PrecisionTrait<PREC_TO>::value_type* dst,
                            const typename PrecisionTrait<PREC_FROM>::value_type* src, size_t nelem) {
+    using src_type = typename PrecisionTrait<PREC_FROM>::value_type;
     using dst_type = typename PrecisionTrait<PREC_TO>::value_type;
 
     for (size_t i = 0; i < nelem; i++) {
         dst[i] = static_cast<dst_type>(src[i]);
+        if (static_cast<src_type>(dst[i]) != src[i] && // Narrowing conversion
+                src[i] == std::numeric_limits<src_type>::max()) {
```

Comment:
Why not just check that value is out of bounds for I32 and clamp it?

Reply:
Actually the main idea was to solve not the whole problem of narrowing conversion, but just the one case when we have `<int64_t>::max()` -> `<int64_t>::max()` but as was discussed with @ilya-lavrenov it can also be solved by introducing the analogue of `saturate_cast` in OpenCV which do the same as you suggest.

Reply:
Anyway don't see a reason to solve particular issue - conversion should be consistent

Reply:
> conversion should be consistent

Which one? By introducing the `saturate_cast` function, we make conversion to be universal since this function will be used in all places where we convert data types.

Reply:
What I mean is we do not need to stick to specific value/case (`std::numeric_limits<std::int64_t>::max()`). Either reuse some other conversion or write simple version here:

```cpp
// It is preudo code, don't use it as is
const auto clamped = std::min(std::max(source, std::numeric_limits<DST_TYPE>::min()), std::numeric_limits<DST_TYPE>::max());
```

Reply:
Done.